spec/memo_wise_spec.rb

```diff
@@ -329,4 +341,146 @@ def public_memowise_method
       expect(instance2.no_args_counter).to eq(1)
     end
   end
+
+  describe "#preset_memo_wise" do
+    it "presets memoization for no args" do
+      instance.preset_memo_wise(:no_args) { "preset_no_args" }
+
+      expect(Array.new(4) { instance.no_args }).to all eq("preset_no_args")
+      expect(instance.no_args_counter).to eq(0)
+    end
+
+    it "presets memoization for methods with positional arguments" do
+      instance.preset_memo_wise(:with_positional_args, 1, 2) { "preset_1_2" }
+      instance.preset_memo_wise(:with_positional_args, 3, 4) { "preset_3_4" }
+
+      expect(Array.new(4) { instance.with_positional_args(1, 2) }).
+        to all eq("preset_1_2")
+
+      expect(Array.new(4) { instance.with_positional_args(3, 4) }).
+        to all eq("preset_3_4")
+
+      expect(instance.with_positional_args_counter).to eq(0)
+    end
+
+    it "presets memoization for methods with keyword arguments" do
+      instance.preset_memo_wise(:with_keyword_args, a: 1, b: 2) { "first" }
+      instance.preset_memo_wise(:with_keyword_args, a: 2, b: 3) { "second" }
+
+      expect(Array.new(4) { instance.with_keyword_args(a: 1, b: 2) }).
+        to all eq("first")
+
+      expect(Array.new(4) { instance.with_keyword_args(a: 2, b: 3) }).
+        to all eq("second")
+
+      expect(instance.with_keyword_args_counter).to eq(0)
+    end
+
+    it "presets memoization for methods with special characters in the name" do
+      instance.preset_memo_wise(:special_chars?) { "preset_special_chars?" }
+      expect(Array.new(4) { instance.special_chars? }).
+        to all eq("preset_special_chars?")
+      expect(instance.special_chars_counter).to eq(0)
+    end
+
+    it "presets memoization for methods set to false values" do
+      instance.preset_memo_wise(:true_method) { false }
+      expect(Array.new(4) { instance.true_method }).to all eq(false)
+      expect(instance.true_method_counter).to eq(0)
+    end
+
+    it "presets memoization for methods set to nil values" do
+      instance.preset_memo_wise(:no_args) { nil }
+      expect(Array.new(4) { instance.no_args }).to all eq(nil)
+      expect(instance.no_args_counter).to eq(0)
+    end
+
+    context "when there is already a memoized value" do
```

Comment:
Since these are basically the same tests as the above (except for the first and last lines), I'm wondering if there's an easy way to use a `shared_examples` block for both of these. Thoughts?